pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -802,3 +803,8 @@ func (m *ManagerImpl) isDevicePluginResource(resource string) bool {
 	}
 	return false
 }
+
+// GetDevices returns the devices used by the specified container
+func (m *ManagerImpl) GetDevices(podUID, containerName string) []*podresourcesapi.ContainerDevices {
```

Comment:
This is the kind of issue that a mutex at the podDevices struct level would have prevented.
Code becomes pretty hard to maintain if it's not carefully reviewed with the mutex decision to make it "global".

You'll have to call `m.mutex.Lock()` and `m.mutex.Unlock()` here

Reply:
good catch, done.